### ISSUE TITLE
Cannot get the textgrid result when the input_folder only has only one pair of data(one wav and corresponding lab file)

### ISSUE THREAD

**Issue by quanzzz123** [MontrealCorpusTools/Montreal-Forced-Aligner#178]
Hello, thanks for the great job!
Here comes one problem in my case. I just want do forced alignment on only one input .wav file with .lab, however there is no textgrid output in the output folder.
And I test some other cases. If I put two different pairs of data( different name and different .wav), I will get the correct result.
If I use the same pair of data with different names, I will not get any result.
So I go to check the logs.
In the "align.0.0.log" comes the error as shown below:

LOG (gmm-boost-silence[5.4.251~1-094d2]:main():gmm-boost-silence.cc:103) Wrote model to -
LOG (gmm-align-compiled[5.4.251~1-094d2]:main():gmm-align-compiled.cc:127) 01
WARNING (gmm-align-compiled[5.4.251~1-094d2]:AlignUtteranceWrapper():decoder-wrappers.cc:466) Retrying utterance 01 with beam 40
WARNING (gmm-align-compiled[5.4.251~1-094d2]:AlignUtteranceWrapper():decoder-wrappers.cc:475) Did not successfully decode file 01, len = 361
LOG (gmm-align-compiled[5.4.251~1-094d2]:main():gmm-align-compiled.cc:135) Overall log-likelihood per frame is nan over 0 frames.
LOG (gmm-align-compiled[5.4.251~1-094d2]:main():gmm-align-compiled.cc:137) Retried 1 out of 1 utterances.
LOG (gmm-align-compiled[5.4.251~1-094d2]:main():gmm-align-compiled.cc:139) Done 0, errors on 1

It seems like when using beam 40 , there is something wrong with retrying utterance.
For the two different pair of data case, there is no WARNING information.

So, can someone help me with this? Also I will go to the source code to try to fix it.

**Comment by simlmx** [comment]
I had the same issue. Making sure all the words are in the dictionary fixed it for me.

**Comment by quanzzz123** [comment]
> I had the same issue. Making sure all the words are in the dictionary fixed it for me (_i.e._ no prompt to fix words not in the dictionary and an empty `oovs_found.txt` file).
> I think it can tolerate some out-of-vocabulary words but I don't know what the threshold is.

Thanks.
Actually the words of my lab file and wav all exist in the dictionary,  maybe its the problem of Kaldi align shell scripts? 
I have no idea of it.

**Comment by simlmx** [comment]
@quanzzz123 Yeah turns out I encountered the same problem (no output) at times where all my words are in the vocabulary. No idea what is going on either!

**Comment by GalaxieT** [comment]
Same problem. Turns out that I was using some wrong lab file. It seems that things would go will if there weren't so many words out of the dictionary.
